org.omg.sysml/src/org/omg/sysml/util/repository/EMFModelRefresher.java

```diff
@@ -0,0 +1,257 @@
+			//so the importedElement UUID is used instead.
+			Object importedMembers = dto.get("importedElement");
+			EObject resolvedReference = resolveReference(importedMembers, false);
+			if (resolvedReference != null) {
+				membershipImport.setImportedMembership(((org.omg.sysml.lang.sysml.Element) resolvedReference).getOwningMembership());
+			}
+		}
+	}
+
+	private boolean canTransformInternals(EObject langElement) {
+		return isNotStandardLibraryElement(langElement);
+	}
+	
+	private EClass getTypeClass(Object typeField) {
+		if (typeField instanceof String typeName) {
+			return (EClass) SysMLPackage.eINSTANCE.getEClassifier(typeName); 
+		}
+		return null;
+	}
+	
+	private boolean isNotStandardLibraryElement(EObject langElement) {
+		return !ElementUtil.isStandardLibraryElement((org.omg.sysml.lang.sysml.Element) langElement);
+	}
+
+	private void transformContainmentReferences(EObject langElement, Element remoteElement) {
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.get(feature.getName()) != null) {
+				if (feature instanceof EReference reference && reference.isContainment()) {
+					transformStructuralFeature(langElement, remoteElement, reference, true, true);
+				}
+			}
+		}
+	}
+	
+	private void tranformCrossreferences(EObject langElement, Element remoteElement) {
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.containsKey(feature.getName())) {
+				if (feature instanceof EReference reference) {
+					transformStructuralFeature(langElement, remoteElement, reference, true, false);
+				}
+			}
+		}
+		
+		if (langElement instanceof FeatureTyping ft) {
+			ft.setTypedFeature(null);
+		}
+	}
+	
+	private void tranformAttributes(EObject langElement, Element remoteElement) {
+		for (EStructuralFeature feature: getStructuralFeatuers(langElement))
+		{
+			if (!feature.isDerived() && remoteElement.containsKey(feature.getName())) {
+				if (feature instanceof EAttribute reference) {
+					transformStructuralFeature(langElement, remoteElement, reference, false, false);
+				}
+			}
+		}
+	}
+	
+	private static Collection<EStructuralFeature> getStructuralFeatuers(EObject langElement){
+		EClass eClass = langElement.eClass();
+		
+		Set<EObject> allRedefined = eClass.getEAllStructuralFeatures().stream()
+				.map(f -> f.getEAnnotation("redefines")).filter(Objects::nonNull)
+				.flatMap(a -> a.getReferences().stream()).collect(Collectors.toSet());
+		
+		return eClass.getEAllStructuralFeatures().stream().filter(f -> !allRedefined.contains(f)).collect(Collectors.toList());
+	}
+	
+	@SuppressWarnings("unchecked")
+	private void transformStructuralFeature(EObject langElement,  Element remoteElement, EStructuralFeature feature, boolean isReference, boolean isContainment) {
+		if (!disabledStructuralFeatures.contains(feature.getName())) {

```

Comment:
Please review whether this can be simplified in some way, the very deep nesting makes this code very hard to read.

Reply:
Handled in #616